Change asset scale to 9, gwei

kincaidoneil opened the pull request:
- Remove bandwidth config option (should be implemented in connector, not here)
- Throw error if no money/data handler is registered per LPI v2 spec
- Debug logs now use eth, not wei/gwei
- Add class for simpler unit conversions

michielbdejong commented:
Because of your comment at https://github.com/interledgerjs/ilp-plugin-ethereum/issues/12#issuecomment-327966015 I made it so you can also choose not to pass in the secret, and then it will not try to unlock the account.

kincaidoneil committed:
Change asset scale to 9, gwei

- Remove bandwidth config option (should be implemented in connector, not here)
- Throw error if no money/data handler is registered per LPI v2 spec
- Debug logs now use eth, not wei/gwei
- Add class for simpler unit conversions

emschwartz commented on index.js:
```diff
@@ -135,31 +139,31 @@ class Plugin extends PluginMiniAccounts {
-      throw new Error('Insufficient bandwidth, used: ' + unsecured + ' max: ' +
-        this._bandwidth)
-    }
+    // TODO: Fix this !
```
?

emschwartz commented on index.js:
```diff
@@ -135,31 +139,31 @@ class Plugin extends PluginMiniAccounts {
-      secured.toString(), 'prepared amount', amount, 'newPrepared',
-      newPrepared.toString(), 'prepared', prepared.toString())
 
-    if (unsecured.greaterThan(this._bandwidth)) {
```
I don't think the connector can set this, because it doesn't track the unsecured per account in the multi-plugin

sharafian commented on index.js:
```diff
@@ -27,16 +27,13 @@ class Plugin extends PluginMiniAccounts {
     this._db = opts.db || 'machinomy_db'
     this._provider = opts.provider || 'http://localhost:8545'
-    this._minimumChannelAmount = new BigNumber(opts.minimumChannelAmount || 100)
+    this._minimumChannelAmount = new EthUnit(opts.minimumChannelAmount || 1, 'gwei')
```
If this is the smallest divisible unit, shouldn't the channel capacity be more than 1?

kincaidoneil replied:
Yeah... that doesn't make sense. What do you think of using 1000000 gwei / 0.001 ETH? ~$0.40 at currency exchange rate, so about the same as xrp-asym-server's default of 1 XRP

sharafian replied:
:+1:

sharafian commented on index.js:
```diff
@@ -100,7 +97,14 @@ class Plugin extends PluginMiniAccounts {
+          }
+
+          if (amount !== '0') {
+            this._moneyHandler(amount)
```
it's more efficient to just configure the connector not to track this plugin's balance than it is to pretend a settlement came in on each packet

kincaidoneil replied:
Could you elaborate a little more on what you mean by pretending a settlement came in? I don't have a full understanding of how the connector's balance logic interacts with this

sharafian replied:
if you call `moneyHandler`, then the connector will increase their balance for this plugin. When they send a packet out on this plugin, they'll decrease the balance. But here, `moneyHandler` is just being called after a fulfill, not after a settlement. It means that for every packet, the balance gets reset to zero. To achieve the same effect, you can just configure the connector to not track a balance at all for this plugin

sharafian commented on index.js:
```diff
@@ -135,31 +139,31 @@ class Plugin extends PluginMiniAccounts {
-      newPrepared.toString(), 'prepared', prepared.toString())
 
-    if (unsecured.greaterThan(this._bandwidth)) {
-      throw new Error('Insufficient bandwidth, used: ' + unsecured + ' max: ' +
```
We should add a maxPacketAmount like in ilp-plugin-xrp-asym server. It's important for supporting ILP protocol STREAM

kincaidoneil replied:
Reading through xrp, this makes a lot more sense now. (As it turns out, `_handleIncomingPrepare` was never even called to begin with!). As a side note, would there be a way to extend ilp-plugin-mini-balances to abstract away the balance logic (somewhat)? Looks like it'd need to be more extensible in handling custom BTP messages, but just curious where you envisioned that plugin being used or if it's just better for each server plugin to roll its own balance logic like xrp-asym-server.

sharafian replied:
ILP Plugin Payment (https://github.com/interledgerjs/ilp-plugin-payment) is pretty much whatt you're asking for. It's still rough around the edges but should be usable once https://github.com/interledgerjs/ilp-plugin-payment/pull/1 this is merged. It includes an example of how it could be used to create an XRP payment based plugin, so you could start off of that

kincaidoneil replied:
Thanks. Using ilp-plugin-payment for an on-ledger integration makes sense to me, but I'm a little confused in this case though, how would the paychan updates be exchanged between client and server with it?

kincaidoneil commented:
Superceded by #14.
